### PR TITLE
Add support for http status code as a field

### LINKED ISSUE #1569: The HTTP Post node should allow logging on HTTP error codes
Kapacitor should expose information about the response it receives after making the POST request.

* If the HTTP error code is not a 2xx, then the error code should be logged.
* Optionally the entire response body should be logged, if the error code is not 2xx
* Optionally the error code should be placed as a new field on the data.

@desa  Thoughts?

### PR DESCRIPTION
**Pull request by nathanielc**
Fixes #1569

- [x] Rebased/mergable
- [x] Tests pass
- [x] CHANGELOG.md updated

### CODE REVIEW

File: http_post.go [hunk 1]
```diff
@@ -108,30 +134,54 @@ func (g *httpPostGroup) DeleteGroup(d edge.DeleteGroupMessage) (edge.Message, er
...
+			var body []byte
+			body, err = ioutil.ReadAll(resp.Body)
+			if err == nil {
+				// Use the body content as the error
+				err = errors.New(string(body))
+			}
+		} else {
+			err = errors.New("unknown error, use .captureResponse() to capture the HTTP response")
+		}
+		n.diag.Error("POST returned non 2xx status code", err, keyvalue.KV("code", strconv.Itoa(resp.StatusCode)))
```
Comment (desa) [inline http_post.go]: I just now understood what you were saying here https://github.com/influxdata/kapacitor/pull/1535#discussion_r139201996 about making everything strings.

File: services/httppost/service.go [hunk 2]
```diff
@@ -310,5 +316,21 @@ func (h *handler) Handle(event alert.Event) {
...
+			var body []byte
+			body, err = ioutil.ReadAll(resp.Body)
+			if err == nil {
+				// Use the body content as the error
+				err = errors.New(string(body))
+			}
+		} else {
+			err = errors.New("unknown error, use .captureResponse() to capture the HTTP response")
+		}
+		h.diag.WithContext(keyvalue.KV("code", strconv.Itoa(resp.StatusCode))).Error("POST returned non 2xx status code", err)
```
Comment (desa) [inline services/httppost/service.go]: `WithContext` returns a new diagnostic. Might be worth it to change the `Error` method on `Diagnostic` to

```go
Error(msg string, err error, ctx ...keyvalue.T)
```

Comment (nathanielc) [reply services/httppost/service.go]: will do

Comment (nathanielc) [reply services/httppost/service.go]: Fixed

**Pull request merged**